src/com/epam/javatrainings/Validator.java

```diff
@@ -0,0 +1,76 @@
+package com.epam.javatrainings;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.regex.*;
+import java.time.Period;
+
+public class Validator <T> {
+    public static final Pattern VALID_EMAIL_ADDRESS_REGEX =  Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+    private ArrayList<Error> errorList;
+    private T dto;
+
+    public Validator(T dto){
+        this.dto = dto;
+        errorList = new ArrayList<Error>();
+    }
```

Comment:
Need to create only one Validator instance which will validate all objects, 
so redesign the class so this method can take obj, validate it, and if it has any errors provide them